### DISCUSSION TITLE
GOG - Civilization IV - innoextract failed to extract GOG setup file

### DISCUSSION THREAD

**Discussion by Zoldronesk** [lutris/lutris#5812] [General]
innoextract failed to extract GOG setup file

multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/lib/python3.11/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
                    ^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/lutris/util/extract.py", line 31, in extract_archive
    _do_extract(path, temp_path, opener, mode, extractor)
  File "/app/lib/python3.11/site-packages/lutris/util/extract.py", line 144, in _do_extract
    _extract_gog(archive, dest)
  File "/app/lib/python3.11/site-packages/lutris/util/extract.py", line 231, in _extract_gog
    _decompress_gog(path, dest)
  File "/app/lib/python3.11/site-packages/lutris/util/extract.py", line 213, in _decompress_gog
    raise RuntimeError("innoextract failed to extract GOG setup file")
RuntimeError: innoextract failed to extract GOG setup file
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):

  File "/app/lib/python3.11/site-packages/lutris/util/jobs.py", line 30, in target
    result = self.function(*a, **kw)
             ^^^^^^^^^^^^^^^^^^^^^^^

  File "/app/lib/python3.11/site-packages/lutris/installer/interpreter.py", line 325, in dispatch
    return method(params)
           ^^^^^^^^^^^^^^

  File "/app/lib/python3.11/site-packages/lutris/installer/commands.py", line 212, in extract
    self._killable_process(extract.extract_archive, filename, dest_path, merge_single, extractor)

  File "/app/lib/python3.11/site-packages/lutris/installer/commands.py", line 548, in _killable_process
    result = result_obj.get()  # Wait process end & re-raise exceptions
             ^^^^^^^^^^^^^^^^

  File "/usr/lib/python3.11/multiprocessing/pool.py", line 774, in get
    raise self._value

RuntimeError: innoextract failed to extract GOG setup file

Lutris log:
[INFO:2024-12-20 22:21:14,644:application]: Starting Lutris 0.5.18
[INFO:2024-12-20 22:21:14,746:startup]: "card1" is NVIDIA GeForce GTX 660 Ti (10de:1183 3842:3660 nvidia) Driver 470.256.02
[ERROR:2024-12-20 22:21:14,752:system]: ['vulkaninfo', '--summary'] command failed: Command '['vulkaninfo', '--summary']' returned non-zero exit status 1.
[INFO:2024-12-20 22:21:14,766:startup]: "card0" is No GPU (simple-framebuffer) Driver None
[WARNING:2024-12-20 22:21:14,832:startup]: Vulkan reports that the 'NVIDIA GeForce GTX 660 Ti' device has API version of 1.2.175. 1.3.0 is required for the latest DXVK.
[INFO:2024-12-20 22:21:51,891:gog]: Getting game details for 1760534591
[INFO:2024-12-20 22:21:54,700:gog]: Getting game details for 1760534591
[INFO:2024-12-20 22:22:05,865:installer]: Getting files for gogsetup
[INFO:2024-12-20 22:22:05,865:gog]: Getting game details for 1760534591
[INFO:2024-12-20 22:22:07,181:gog]: Getting download info for https://api.gog.com/products/1760534591/downlink/installer/en1installer3
[INFO:2024-12-20 22:22:08,446:installer]: Getting files for gogsetup
[INFO:2024-12-20 22:22:08,447:gog]: Getting game details for 1760534591
[INFO:2024-12-20 22:22:08,544:gog]: Getting download info for https://api.gog.com/products/1760534591/downlink/installer/en1installer4
[INFO:2024-12-20 22:22:09,260:gog]: Getting download info for https://api.gog.com/products/1760534591/downlink/installer/en1installer3
[INFO:2024-12-20 22:22:10,768:gog]: Getting download info for https://api.gog.com/products/1760534591/downlink/installer/en1installer4
[INFO:2024-12-20 22:22:16,115:file_box]: Download completed
[WARNING:2024-12-20 22:24:09,257:system]: Non existent path: /home/zoldron/.var/app/net.lutris.Lutris/data/lutris/runners/proton/ge-proton
[INFO:2024-12-20 22:43:22,697:file_box]: Download completed
[INFO:2024-12-20 22:43:22,699:files_box]: All files available
[INFO:2024-12-20 22:43:22,699:installerwindow]: All files are available, continuing install
[INFO:2024-12-20 22:43:22,700:installerwindow]: Launching installer commands
[INFO:2024-12-20 22:43:22,850:wine]: Creating a win64 prefix in /home/zoldron/Games/gog/sid-meiers-civilization-iv
[INFO:2024-12-20 22:43:22,850:wine]: Winepath: /home/zoldron/.var/app/net.lutris.Lutris/data/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine
[INFO:2024-12-20 22:43:35,030:wine]: win64 Prefix created in /home/zoldron/Games/gog/sid-meiers-civilization-iv
[ERROR:2024-12-20 22:43:35,172:jobs]: Error while completing task <function ScriptInterpreter._iter_commands.<locals>.dispatch at 0x717e72ed7d80>: <class 'RuntimeError'> innoextract failed to extract GOG setup file
[ERROR:2024-12-20 22:43:35,175:interpreter]: Last install command failed, show error

**Comment by danieljohnson2** [comment]
Not seeing the answer in your logs. I see that there are about forty seven Civilization installers. Can you tell us which installer your chose? As in "GOG Windows", "GOG (auto-generated)" or just "GOG", or something else?

**Reply by Zoldronesk** [reply]
I only see two, auto-generatd and this one which I<m trying to install:

 Lutris script maintained by legluondunet: https://github.com/legluondunet/MyLittleLutrisScripts

**Reply by danieljohnson2** [reply]
Have you tried the auto-generated one? I find they work well for me.

**Reply by Zoldronesk** [reply]
Yes they work, the one bugging is for a mod pack.

**Comment by danieljohnson2** [comment]
OK, I am not too sure what is wrong, but I think the thing to do is to do a better job of capturing the error output of Innoextract and see what we see. So I've upgraded the latest Lutris master to do this, and I'd like you to try it (if you're even still interested in this- I know I took a while to get back to you!)

I'd like you to [download the latest master](https://github.com/lutris/lutris) and then run the ``bin/lutris`` script within- no need to install. Just try to run the installer and provide us with the log output. There should be an additional line about Innoextract in there and I am hoping that will provide a clue.
